.github/workflows/ubuntu.yml

```diff
@@ -14,16 +14,18 @@ jobs:
         fetch-depth: 0
         submodules: true
     - name: Install APT packages
+      # Keep in sync with "Prerequisites" in User's Guide.
       run: |
         sudo apt-get update
         sudo apt install libblosc-dev libbz2-dev libhdf5-dev liblz4-dev liblzo2-dev libsnappy-dev libzstd-dev zlib1g-dev
-        sudo apt install python3-all-dev python3-setuptools python3-packaging python3-numexpr  # python3-numpy cython3
+        sudo apt install python3-all-dev python3-setuptools python3-packaging  # python3-numexpr python3-numpy cython3
```

Comment:
As you prefer.  This *should* work well with latest ubuntu.

Reply:
Thanks @avalentino. I commented out NumExpr from that list because I added it back to `requirements.txt` in d852177, and packages in that file are installed in the following "Install dependencies" step (for that same reason, I could also comment out `python3-packaging` in the command).

However, if that file is instead supposed to only contain *build* dependencies, I can add an explicit comment in there and undo the change (although from <https://pip.pypa.io/en/stable/reference/requirements-file-format/> it looks more like runtime dependencies… maybe the spec is a little blurry there).

Reply:
I'd suggest to keep using requirements.txt for just the `runtime` dependencies whereas the `build` deps could go either hardcoded in CI scripts or in a new requirements-build.txt.  With that, `cython` and `packaging` should be removed from requirements.txt

Reply:
In the normal case (installation via pip), in principle with the adoption of the `pyproject.toml` we should no longer need external `requirements*.txt` files (build time and runtime). I see that we use them in CI so probably it makes sense to keep them.
For the specific `ubuntu.yml` test, in my understanding, the purpose is to test as much as possible how PyTables can play well with the system packages provided by ubuntu (specially the C libraries for HDF5 and compressors).
This is why I put my comment.
Of course, if there are specific limitations with the system packages, we can always install an version compatible with PyTables requirements using pip, but, as I said, the spirit of the test should be to use as much as possible system packages.

Reply:
Thanks @avalentino for the clarification, I think I better understand the purpose of these jobs now. :slightly_smiling_face: In commit ad6e6c1 I did my best to clarify the situation, try to use system packages whenever possible, and make it easy to tell the origin of a certain package.

Reply:
As per commit f5cfbdc, I think that the situation with Ubuntu CI jobs is stable.

Reply:
Thanks @ivilata for the fantastic job!